src/lib.rs

```diff
@@ -585,6 +585,25 @@ pub mod rkyv {
     pub use crate::weekday::ArchivedWeekday;
 }
 
+/// The number of nanoseconds in a microsecond.
+pub const NANOS_PER_MICRO: i32 = 1000;
+/// The number of nanoseconds in a millisecond.
+pub const NANOS_PER_MILLI: i32 = 1_000_000;
+/// The number of nanoseconds in a second.
+pub const NANOS_PER_SEC: i32 = 1_000_000_000;
+/// The number of microseconds in a second.
+pub const MICROS_PER_SEC: i64 = 1_000_000;
```

Comment:
The types of these constants are 'whatever needs the fewest casts in our current code'.

That is: everything that has to do with seconds or more is using the type of the `Duration.secs` field.
Everything that has to do with subsecond units uses the type of the `Duration.nanos` field.
For most constants that seems like a good default.

`NANOS_PER_SEC` and `MICROS_PER_SEC` are at the boundary. `NANOS_PER_SEC` is currently an `i32`, and we sometimes cast it to an `i64`.

Maybe we should change `MICROS_PER_SEC` to also be an `i32`. Argument is consistency with `NANOS_PER_SEC`, and that widening casts are infallible and available with a `From` impl (even though this is just a constant and would not fail anyway.).